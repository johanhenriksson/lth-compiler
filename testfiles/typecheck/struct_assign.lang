struct ex_struct { }

struct other_struct { }

int main() {
    ex_struct a;
    ex_struct b;
    other_struct c;

    a = b; // ok

    a = 5; // type error
    a = true; // type error
    a = c; // type error
}
