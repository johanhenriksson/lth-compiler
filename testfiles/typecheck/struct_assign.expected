Error at line 12,5: cannot assign 'a' of type struct ex_struct, got int
Error at line 13,5: cannot assign 'a' of type struct ex_struct, got bool
Error at line 14,5: cannot assign 'a' of type struct ex_struct, got struct other_struct
